src/test/groovy/org/kohsuke/groovy/sandbox/TheTest.groovy

```diff
@@ -869,4 +869,16 @@ def s = []
 s.join(' ')
 """)
     }
+
+    @Issue("JENKINS-50380")
+    void testCheckedCastWhenAssignable() {
+        assertIntercept(['new NonArrayConstructorList(Boolean,Boolean)',
+                         'NonArrayConstructorList.join(String)'],
```

Comment:
So what did this intercept before?

Reply:
`new NonArrayConstructorList(Boolean,Boolean)` followed by `new NonArrayConstructorList(String)`, which is where it blows up with an `unclassified new ...`